src/tooling/grain_doc.md

```diff
@@ -0,0 +1,148 @@
+
+## Attributes
+
+### @param
+
+```gr
+/**
+ * @param name: description
+ */
+```
+The `@param` attribute provides information for one parameter of the function being documented.
+
+Multiple `@param` attributes should be specified in order of the function parameters. The name specified before the colon (`:`) should match the parameter name exactly—while not currently enforced, it will be required for named parameters in the future. The content after the colon is the description of the parameter.
+
+When generating docs, the type of the parameter will be detected by the compiler and added to the parameter's row in the "Parameters" table.
+
+### @returns
+
+```gr
+/**
+ * @returns description
+ */
+```
+
+The `@returns` attribute provides information for the return value of the function being documented.
+
+This attribute requires a description of the return value. In the Grain standard library, we don't use the `@returns` attribute if the function returns `Void`.
+
+When generating docs, the type of the return value will be detected by the compiler and added to the row in the "Returns" table.
+
+### @example
+
+```gr
+/**
+ * @example code
+ */
+```
+
+The `@example` attribute provides snippets showing how the documented code is used. Currently, only single line examples are allowed.
+
+When generating docs, the code snippet will be wrapped in a code block tagged for the `grain` language and added to an "Examples" section.
+
+### @since
+
+```gr
+/**
+ * @since version
+ */
+```
+
+The `@since` attribute provides the version in which the documented code was added.
+
+This attribute requires a [Semantic Version](https://semver.org/), optionally prefixed with `v`. For example,`v0.2.0` or `0.2.0` are treated the same.
+
+When generating docs, a `<details>` element titled "Added in X.X.X" will be added above the type signature. When this attribute is specified, you'll be required to specify the `--current-version=X.X.X` flag to the `grain doc` command. This allows you to specify newer versions in your source code, and the generated docs will show "Added in next", instead of the version.
+
+### @history
+
+```gr
+/**
+ * @history version: description
+ */
+```
+
+The `@history` attribute provides details on significant changes to the documented code.
+
+This attribute requires a [Semantic Version](https://semver.org/) before the colon (`:`) and a description of the change after the colon.
+
+When generating docs, adds a history table to the `<details>` element created by the `@since` attribute if both were used; otherwise, a new `<details>` element titled  "History" will be inserted above the type signature. When using this attribute, you'll be required to specify the `--current-version=X.X.X` flag to the `grain doc` command. This allows you to specify newer versions in your source code, and the generated docs will show "next" instead of the version.
```

Comment:
```suggestion
When generating docs, adds a history table to the `<details>` element created by the `@since` attribute if both were used; otherwise, a new `<details>` element titled  "History" will be added above the type signature. When using this attribute, you'll be required to specify the `--current-version=X.X.X` flag to the `grain doc` command. This allows you to specify newer versions in your source code, and the generated docs will show "next" instead of the version.
```